src/Services/PatientService.php

```diff
@@ -35,14 +36,24 @@ public function __construct()
     {
     }
 
-    public function validate($patient)
+    public function validate($patient, $context)
     {
         $validator = new Validator();
-
-        $validator->required('fname')->lengthBetween(2, 255);
-        $validator->required('lname')->lengthBetween(2, 255);
-        $validator->required('sex')->lengthBetween(4, 30);
-        $validator->required('DOB')->datetime('Y-m-d');
+        switch ($context) {
+            case 'update':
+                $type = 'optional';
+                break;
+            case 'insert':
+                $type = 'required';
+                break;
+            default:
+                $type = 'required';
+                break;
+        }
+        $validator->$type('fname')->lengthBetween(2, 255);
+        $validator->$type('lname')->lengthBetween(2, 255);
+        $validator->$type('sex')->lengthBetween(4, 30);
+        $validator->$type('DOB')->datetime('Y-m-d');
 
```

Comment:
But isn't a pid required on the update?
Also note that somebody will likely be building this out to add more optional validations for the insert (and the update) and to play nicely with the current layout engine. Thus may not make sense to make it always optional for update and required for insert.